owltorrent-network/src/main/java/edu/rice/owltorrent/network/ClientHandler.java

```diff
@@ -34,23 +31,27 @@ public void run() {
       in = new DataInputStream(socket.getInputStream());
 
       byte[] handShakeBuffer = new byte[68];
-      // TODO: handle read != 68 bytes
-      in.read(handShakeBuffer);
-      Optional<Peer> peer = verifyHandShake(handShakeBuffer);
-      if (peer.isEmpty()) {
+      if (in.read(handShakeBuffer) != 68) {
+        log.info("Received invalid handshake.");
```

Comment:
in the case of invalid handshake, probably want to log error, and also return instead.